packages/Search/src/details/DTK_DetailsTraits.hpp

```diff
@@ -48,6 +48,23 @@ struct Access<View, typename std::enable_if<Kokkos::is_view<View>::value &&
     using Tag = typename Tag<typename View::value_type>::type;
 };
 
+template <typename View>
+struct Access<View, typename std::enable_if<Kokkos::is_view<View>::value &&
+                                            View::Rank == 2>::type>
+{
+    KOKKOS_FUNCTION static Point get( View const &v, int i )
+    {
+        return {v( i, 0 ), v( i, 1 ), v( i, 2 )};
+    }
+
+    static typename View::size_type size( View const &v )
```

Comment:
I think that the size of a 2D View is not well defined. For example, in numpy size is `extent(0)*extent(1)`. This is also what is done in Kokkos https://github.com/kokkos/kokkos/blob/master/core/src/impl/Kokkos_ViewMapping.hpp#L956 Maybe call it something like `extent_0`?

Reply:
This is the common terminology to obtain the number of elements in a container or a range in the standard library.  Extent is a multidimensional array thing.

Reply:
But the number of elements in a 2D view (which is what you have) is `extent(0)*extent(1)` not `extent(0)`.

Reply:
I think depends on terminology. In this case 2D View is simply an implementation of a container of elements of size 3.

Reply:
Access traits do not know nor care we are working with MD arrays.  They translate user data into something the tree constructor can understand.  The tree is only concerned with how many primitives need to be indexed.  It is unambiguous.

Reply:
It's only unambiguous because you know how you want to use it. If you just look at the file itself or include it somewhere else than in the tree construction then it is a lot less clear what it means.